x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality_file_uploader/components/result_step.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiHorizontalRule,
+  EuiSpacer,
+  useEuiTheme,
+} from '@elastic/eui';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import type { AssetCriticalityCsvUploadResponse } from '../../../../../common/entity_analytics/asset_criticality/types';
+
+export const AssetCriticalityResultStep: React.FC<{
+  result?: AssetCriticalityCsvUploadResponse;
+  validLinesAsText: string;
+  errorMessage?: string;
+  onReturn: () => void;
+}> = React.memo(({ result, validLinesAsText, errorMessage, onReturn }) => {
+  const { euiTheme } = useEuiTheme();
+
+  if (result === undefined || errorMessage !== undefined) {
+    return (
+      <>
+        <EuiCallOut
+          data-test-subj="asset-criticality-result-step-error"
+          title={
+            <FormattedMessage
+              defaultMessage="Asset criticality assignment failed."
+              id="xpack.securitySolution.entityAnalytics.assetCriticalityResultStep.errorMessage"
+            />
+          }
+          color="danger"
+          iconType="error"
+        >
+          {errorMessage && errorMessage}
```

Comment:
this is surely not correct? 😅

Reply:
My typescript is rusty, but even though it looks weird isn't this correct? 😅  `errorMessage` is optional, so line 45 says "render `errorMessage` only if it's defined", right?

Reply:
I refactored the component a little bit, and now the check isn't.